addons/account_edi_ubl_bis3/models/account_edi_format.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.exceptions import UserError
+from odoo.tests.common import Form
+
+
+COUNTRY_EAS = {
+    'HU': 9910,
+
+    'AD': 9922,
+    'AL': 9923,
+    'BA': 9924,
+    'BE': 9925,
+    'BG': 9926,
+    'CH': 9927,
+    'CY': 9928,
+    'CZ': 9929,
+    'DE': 9930,
+    'EE': 9931,
+    'UK': 9932,
+    'GR': 9933,
+    'HR': 9934,
+    'IE': 9935,
+    'LI': 9936,
+    'LT': 9937,
+    'LU': 9938,
+    'LV': 9939,
+    'MC': 9940,
+    'ME': 9941,
+    'MK': 9942,
+    'MT': 9943,
+    'NL': 9944,
+    'PL': 9945,
+    'PT': 9946,
+    'RO': 9947,
+    'RS': 9948,
+    'SI': 9949,
+    'SK': 9950,
+    'SM': 9951,
+    'TR': 9952,
+    'VA': 9953,
+
+    'SE': 9955,
+
+    'FR': 9957
+}
+
+
+class AccountEdiFormat(models.Model):
+    ''' This edi_format is "abstract" meaning that it provides an additional layer for similar edi_format (formats
+    deriving from EN16931) that share some functionalities but needs to be extended to be used.
+    '''
+    _inherit = 'account.edi.format'
+
+    ####################################################
+    # Export
+    ####################################################
+
+    def _get_bis3_values(self, invoice):
+        values = super()._get_ubl_values(invoice)
+        values.update({
+            'customization_id': 'urn:cen.eu:en16931:2017#compliant#urn:fdc:peppol.eu:2017:poacc:billing:3.0',
+            'profile_id': 'urn:fdc:peppol.eu:2017:poacc:billing:01:1.0',
+        })
+
+        all_tax_detail_per_tax = {}
+        for line_vals in values['invoice_line_vals_list']:
+            line_vals['price_subtotal_with_no_tax_closing'] = line_vals['line'].price_subtotal
```

Comment:
Why not have it right from the line_vals then?  In generic it is used a certain way and in bis3 differently?

Reply:
In generic it's not used, this is something specific to BIS3